test/http-api.test.ts

```diff
@@ -16,6 +18,95 @@ describe('HTTP API', () => {
     loginUrl,
   });
 
+  describe('network retry', () => {
```

Comment:
Might be useful to have a test that tests that the retries stop after the max is reached